Lib/test/test_capi/test_bytes.py

```diff
@@ -247,6 +247,29 @@ def test_resize(self):
         # CRASHES resize(NULL, 0, False)
         # CRASHES resize(NULL, 3, False)
 
+    def test_join(self):
```

Comment:
Add tests with iterators (`iter([...])`).

Add test with an iterable containing bytearray or memoryview.

Add test with bytearray as the joiner.

Add test with non-iterable second argument.

Add tests with NULL as the first or the second argument. Or, if they are crashed, add comments, like in other tests.

Reply:
I tried to add all of these tests.